src/app/shell/result/result.component.ts

```diff
@@ -109,6 +132,22 @@ export class ResultComponent implements OnInit, OnDestroy {
     );
   }
 
+  /**
+   * Updates / Appends workshop filter to the URL query string
+   * @private
+   */
+  private setFilterStateURLParams(): void {
+    this.filterState$.pipe(takeUntil(this.destroy$)).subscribe((val) => {
+      this.filterState = val;
+      const filterQueryParams = Util.getFilterStateQuery(this.filterState);
+      if (filterQueryParams === '') {
```

Comment:
what do you think if we return null getFilterStateQuery if there is no such result? and then we can get rid of the code below. leave just this.router.navigate([], { queryParams: { filter: filterQueryParams }, replaceUrl: true });

Reply:
I think that is good idea